media.go

```diff
@@ -0,0 +1,119 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/go-telegram-bot-api/telegram-bot-api"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"sync"
+)
+
+const (
+	mediaDB = "media.json"
+)
+
+// mediaList maps `media urls' -> `Telegram media IDs'.
+type mediaList struct {
+	sync.RWMutex
+	Media map[string]string `json:"media"`
+}
+
+// saveMedia saves media list to mediaDB file.
+func saveMedia(media map[string]string) error {
+	buf, err := json.Marshal(media)
+	if err != nil {
+		return err
+	}
+	datadir, err := dataDir()
+	if err != nil {
+		return err
+	}
+
+	tmpfile, err := ioutil.TempFile(datadir, "temp-media")
```

Comment:
Maybe get all these lines and create a generic function called "SafeWrite" or something like that? I'm sure we'll use this in other places.